### DISCUSSION TITLE
Which Linux distro is supported ?

### DISCUSSION THREAD

**Discussion by BobSquarePants** [evilsocket/opensnitch#595] [Q&A] [answered]
Hi everyone,

I'm currently "using" MX Linux. 

and when I tried to install **python3-opensnitch-ui_1.4.2-1_all.deb**

I got
```
python3-opensnitch-ui depends on python3-setuptools; however:
Package python3-setuptools is not installed.

python3-opensnitch-ui depends on python3-pyqt5.qtsql; however:
Package python3-pyqt5.qtsql is not installed.

python3-opensnitch-ui depends on python3-pyinotify; however:
Package python3-pyinotify is not installed.

python3-opensnitch-ui depends on python3-grpcio; however:
Package python3-grpcio is not installed.

python3-opensnitch-ui depends on python3-protobuf; however:
Package python3-protobuf is not installed.

python3-opensnitch-ui depends on python3-slugify; however:
Package python3-slugify is not installed. 
```

It seem that MX Linux repo have a package for MX 
```
opensnitch-ui/mx,mx 1.3.6-1~mx21+1 all
opensnitch/mx 1.3.6-1~mx21+1 amd64
opensnitch/mx 1.3.6-1~mx21+1 i386
```
But it's based on an older version version of opensnitch :/

Is there a way to install the lastest version of opensnitch (on MX or other) that will install all the necessary decency ? like an apt --install ?

or better a .deb file including all the dependency ? (not sure that it exist :)

Thanks.

**Comment by SwampRabbit** [comment, marked as answer]
@BobSquarePants I have addressed your thread on the MX forums, I will work on getting OpenSnitch updated in our Repos when I can squeeze in the time.  I played cat and mouse with it for a bit until I decided I would put it on hold to keep up with other packages.

It is not really the responsibility of OpenSnitch developers to do what you are asking.  
It is their responsibility to ensure they have a clean release tarball, properly use a standard build system and techniques, provide good release notes and build info like dependency versions.  Those sorts of things.

Obviously the OpenSnitch team puts a lot of time and energy into ensuring as many users from different Distros can use it, but ultimately the packaging is up to the Distro package maintainers.

**Reply by gustavo-iniguez-goya** [reply]
Thank you @SwampRabbit !

> It is their responsibility to ensure they have a clean release tarball, properly use a standard build system and techniques, provide good release notes and build info like dependency versions. Those sorts of things.

If there's something that I can do to improve it just let me know :+1: 

@BobSquarePants I think you already figured it out, after installing it with dpkg, an `apt -f install` should resolve all dependencies.

If I'm not mistaken, `apt install ./python3-opensnitch-ui_1.4.2-1_all.deb` should also resolve the needed deps.

**Reply by SwampRabbit** [reply]
@gustavo-iniguez-goya I quickly started updating the debian/control files, they had some issues which popped warnings during the build, plus minor some things were just incorrect.
https://github.com/SwampRabbit/opensnitch If you want me to do a PR for this I can, not perfect, but its better.

Using the release tarball doesn't seem to build opensnitch-ui package, just the daemon package, I believe that was the case or something similar before.

Someone from Parrot OS and I both had workarounds for this, either combining the debian/control and rules files OR splitting out opensnitch-ui. From an attempt really quickly that combining technique needs updated now.
I can build the daemon and ui separately no problem and there doesn't seem to be any side affects so far.

I'll be messing with it some more in the coming days, or at least I'll try.

**Reply by gustavo-iniguez-goya** [reply]
> If you want me to do a PR for this I can, not perfect, but its better.

ok, both `control` files look fine to me :+1:  

> Using the release tarball doesn't seem to build opensnitch-ui package, just the daemon package, I believe that was the case or something similar before.

@dmknght addressed that problem here: https://github.com/evilsocket/opensnitch/issues/367

I've just tried they recipe and works fine. Here are the files I've used: 
[opensnitch.install.txt](https://github.com/evilsocket/opensnitch/files/7917338/opensnitch.install.txt)
[opensnitch-ui.install.txt](https://github.com/evilsocket/opensnitch/files/7917336/opensnitch-ui.install.txt)
[control.txt](https://github.com/evilsocket/opensnitch/files/7917345/control.txt)
[rules.txt](https://github.com/evilsocket/opensnitch/files/7917357/rules.txt)

I haven't tested the generated debs though, I've only compiled them.

Notes:
- I've included in the rules file, under the override_dh_auto_build section, the needed commands to download the grpc utils.
- I've switched the order of the packages in the control file. Putting opensnitch-ui first packages the files postinst and postrm with the UI deb, whilst it creates new postinst/rm via debhelper for the daemon (I don't know how this works really...)

I'll also delete from the repos the postinstall dialogs to install packages with pip, so we'll also get rid off the whiptail | dialog dependencies by default. I'll create those packages only as a workaround for old distributions that some people still use (ubuntu 16/18).

**Reply by SwampRabbit** [reply]
@gustavo-iniguez-goya wow, you're keeping busy.  :)

Yes that's the Parrot OS person, I remembered that discussion, but was so strapped for time when I posted, and honestly my brain was toast.

I must have messed something up in my local repo build files between version 1.3.6 and now.  I'll try a build with the files you just shared in the AM and let you know.

I did notice quite a bit of stuff to remain backwards compatible for older Distros, which I can understand.  One way to handle the dependencies and even opensnitch itself is to use the openSUSE Build Service since you then can build for a ton of Distros and versions.
We can talk about that later though.  ;)

I did want to ask you about the whole use of pip, glad you were already thinking about it.

Edit: BTW, the build of v1.5.0rc2 I did get done and tested for quite awhile works like a charm!

**Reply by SwampRabbit** [reply]
@gustavo-iniguez-goya this part of the rules file is going to be an issue as it requires the pbuilder chroot to be modified to allow internet access.

```
	go get google.golang.org/protobuf || true
	go install google.golang.org/protobuf/cmd/protoc-gen-go
	go get google.golang.org/grpc/cmd/protoc-gen-go-grpc
	python3 -m pip install --user grpcio-tools
```

I'll have to mess with it some more when I get some free time.

**Reply by gustavo-iniguez-goya** [reply]
oops, you're right. That's the main main problem I had when I tried to compile the packages on opensuse build service.
